demo/src/app/examples/other/cascaded-select/app.component.ts

```diff
@@ -89,4 +91,8 @@ export class AppComponent {
   submit() {
     alert(JSON.stringify(this.model));
   }
+
+  ngOnDestroy(): void {
+    this.onDestroy$.complete();
```

Comment:
🤔 I think this won't be enough. You should rather first `this.onDestroy$.next()`, otherwise the observable stream won't be terminated if I'm not wrong.

See: https://gist.github.com/btroncone/fe9d9aaf457f2ebd72c4624e34d664f8

Reply:
I doubt it, I think `takeUntil` should take account of `complete` too, @benlesh ?

Reply:
I had a similar scenario when switching between my Angular application routes. And by just doing the `.complete()` thing, http requests triggered via observable streams didn't get cancelled. Instead doing a `.next()` worked as expected.

@btroncone can shed some light on this for sure?

Reply:
just tested on an example and seems you were right, I'm going to fix it. thanks!
I often used `complete` without noticing it :sweat_smile: